cluster/images/etcd/migrate/migrate.go

```diff
@@ -0,0 +1,185 @@
+	flags.StringVar(&opts.targetStorage, "target-storage", "", "storage version of etcd to migrate to, one of: etcd2, etcd3")
+	flags.StringVar(&opts.etcdServerArgs, "etcd-server-extra-args", "", "additional etcd server args for starting etcd servers during migration steps, --peer-* TLS cert flags should be added for etcd clusters with more than 1 member that use mutual TLS for peer communication.")
+	migrateCmd.Execute()
+}
+
+// runMigrate validates the command line flags and starts the migration.
+func runMigrate() {
+	if opts.name == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			glog.Errorf("Error while getting hostname to supply default --name: %v", err)
+			os.Exit(1)
+		}
+		opts.name = fmt.Sprintf("etcd-%s", hostname)
+	}
+
+	if opts.ttlKeysDirectory == "" {
+		opts.ttlKeysDirectory = fmt.Sprintf("%s/events", opts.etcdDataPrefix)
+	}
+	if opts.initialCluster == "" {
+		opts.initialCluster = fmt.Sprintf("%s=http://localhost:2380", opts.name)
+	}
+	if opts.targetStorage == "" {
+		glog.Errorf("--target-storage is required")
+		os.Exit(1)
+	}
+	if opts.targetVersion == "" {
+		glog.Errorf("--target-version is required")
+		os.Exit(1)
+	}
+	if opts.dataDir == "" {
+		glog.Errorf("--data-dir is required")
+		os.Exit(1)
+	}
+	if opts.bundledVersionString == "" {
+		glog.Errorf("--bundled-versions is required")
+		os.Exit(1)
+	}
+
+	bundledVersions, err := ParseSupportedVersions(opts.bundledVersionString)
+	if err != nil {
+		glog.Errorf("Failed to parse --supported-versions: %v", err)
+	}
+	err = validateBundledVersions(bundledVersions, opts.binDir)
+	if err != nil {
+		glog.Errorf("Failed to validate that 'etcd-<version>' and 'etcdctl-<version>' binaries exist in --bin-dir '%s' for all --bundled-verions '%s': %v",
+			opts.binDir, opts.bundledVersionString, err)
+		os.Exit(1)
+	}
+
+	target := &EtcdVersionPair{
+		version:        MustParseEtcdVersion(opts.targetVersion),
+		storageVersion: MustParseEtcdStorageVersion(opts.targetStorage),
+	}
+
+	migrate(opts.name, opts.peerListenUrls, opts.peerAdvertiseUrls, opts.binDir, opts.dataDir, opts.etcdDataPrefix, opts.ttlKeysDirectory, opts.initialCluster, target, bundledVersions, opts.etcdServerArgs)
+}
+
+// migrate opens or initializes the etcd data directory, configures the migrator, and starts the migration.
+func migrate(name string, peerListenUrls string, peerAdvertiseUrls string, binPath string, dataDirPath string, etcdDataPrefix string, ttlKeysDirectory string,
+	initialCluster string, target *EtcdVersionPair, bundledVersions SupportedVersions, etcdServerArgs string) {
+
+	dataDir, err := OpenOrCreateDataDirectory(dataDirPath)
+	if err != nil {
+		glog.Errorf("Error opening or creating data directory %s: %v", dataDirPath, err)
+		os.Exit(1)
+	}
+
+	cfg := &EtcdMigrateCfg{
+		binPath:           binPath,
+		name:              name,
+		port:              18629,
```

Comment:
This has to be different for event etcd and for regular etcd - currently you will get the confilct and i guess something will break.